src/core/quantum_gate.jl

```diff
@@ -101,6 +133,106 @@ get_connected_qubits(gate::AbstractGate)=[gate.target]
 
 get_gate_parameters(gate::AbstractGate)=Dict{String,Real}()
 
+"""
+    ControlledGate{G<:AbstractGate}<:AbstractControlledGate
+
+The `ControlledGate` object allows the construction of a controlled `Gate` using a 
+single-target `Operator` (the `kernel`) and the corresponding control and target qubits, 
+specified as `[control_qubit,target_qubit]`.
+
+# Examples
+Whereas a `Hadamard` Gate with `target=1` is constructed by calling `hadamard(target)`, a 
+controlled Hadamard gate is constructed using the `hadamard()` `Function`, and by specifying
+a `connected_qubits=[control, target]`. For instance, the construction 
+of a `ControlledGate{Hadamard}` with `control=1` and `target=2` is performed by calling:
+
+```jldoctest controlled_hadamard
+julia> controlled_hadamard=ControlledGate(hadamard,[1,2])
+Gate Object: ControlledGate{Snowflake.Hadamard}
+Connected_qubits	: [1, 2]
+Operator:
+(4, 4)-element Snowflake.DenseOperator:
+Underlying data ComplexF64:
+1.0 + 0.0im    0.0 + 0.0im    0.0 + 0.0im    0.0 + 0.0im
+0.0 + 0.0im    1.0 + 0.0im    0.0 + 0.0im    0.0 + 0.0im
+0.0 + 0.0im    0.0 + 0.0im    0.7071067811865475 + 0.0im    0.7071067811865475 + 0.0im
+0.0 + 0.0im    0.0 + 0.0im    0.7071067811865475 + 0.0im    -0.7071067811865475 + 0.0im
+
+```
+
+It can then be used in a `QuantumCircuit` as any other `Gate`, and its display symbol is 
+inherited from the display symbol of the single-target `Hadamard` `Gate`:
+
+```jldoctest controlled_hadamard
+julia> circuit=QuantumCircuit(qubit_count=2,gates=[controlled_hadamard])
+Quantum Circuit Object:
+   qubit_count: 2 
+q[1]:──*──
+       |  
+q[2]:──H──
+          
+```
+
+
+"""
+struct ControlledGate{G<:AbstractGate}<:AbstractControlledGate
+    kernel::AbstractOperator
```

Comment:
I think the kernel should be a gate (gate symbol after https://github.com/anyonlabs/Snowflake.jl/pull/232) instead of an operator.

EDIT: I understand why you made it an operator due to how the gate symbols are managed, but if we change that, then, to me, at least, it makes more sense to make it a gate.